### ISSUE TITLE
Missing text-align for image caption

### ISSUE THREAD

**Issue by strarsis** [WordPress/gutenberg#19975]
**Describe the bug**
Image captions have no text-alignment option in Gutenberg editor.

**To reproduce**
1. Insert image block.
2. Add some image block caption.
Notice that the image block caption has no text-alignment options.
Only the image block itself has alignment options.

**Expected behavior**
A text-align option for the image block caption.

**Additional context**
Maybe use image block caption field a fuller editor with more text-related options?
Or extend from a paragraph block/field?

**Comment by phpbits** [comment]
@strarsis You can use [EditorsKit](https://wordpress.org/plugins/block-options/) plugin which has text alignment option for captions. Here's the preview on how the Caption Alignment works :

![Gutenberg  text-align for image caption](https://user-images.githubusercontent.com/3365507/73612732-26655500-4629-11ea-8827-1284bdbce11f.gif)

**Comment by paaljoachim** [comment]
Hi

This issue can be broader then only the image caption. I am thinking of creating a new issue, but since we have this issue I can instead add to it.

Various blocks that use captions:

Image Block:
<img width="711" alt="Screen Shot 2020-02-05 at 22 01 20" src="https://user-images.githubusercontent.com/5323259/73883050-847d7c80-4863-11ea-9eb7-f333f75f19de.png">

Embed Block:
<img width="721" alt="Screen Shot 2020-02-05 at 22 01 49" src="https://user-images.githubusercontent.com/5323259/73883092-995a1000-4863-11ea-93b7-9e4033fc1615.png">

Gallery Block (individual gallery image captions):
<img width="649" alt="Screen Shot 2020-02-05 at 21 36 37" src="https://user-images.githubusercontent.com/5323259/73883115-a2e37800-4863-11ea-8395-42e22722ca53.png">

Gallery Block - gallery caption:
<img width="693" alt="Screen Shot 2020-02-05 at 21 37 39" src="https://user-images.githubusercontent.com/5323259/73883177-c0b0dd00-4863-11ea-969a-d710b621f2e8.png">

The above linked PR @epiqueras worked on was closed. See the discussion in the closed PR.

We basically need the caption alignment added to the caption toolbar.
@ellatrix could you give some advice on this? Thank you.

**Comment by ellatrix** [comment]
The issues with the toolbar has always been there (maybe there's even an issue for it). We've always forced it to centre. I think for these inline toolbars we'll have to detect text alignment and position the popover based on that.

**Comment by paaljoachim** [comment]
I came across this issue:
https://github.com/WordPress/gutenberg/issues/12997

**Comment by paaljoachim** [comment]
@ellatrix How can we move forward? 
In relation to @epiqueras https://github.com/WordPress/gutenberg/pull/20015#issuecomment-582448496 comment 
about having RichText support in place. So that we can add image alignment into the caption toolbar.

**Comment by joyously** [comment]
My opinion: there should not be options to align or italic or color or _anything_ the captions on various blocks, because the theme is styling them universally. There is some color style on some captions already, which should be removed, because they override the theme styles.

**Comment by strarsis** [comment]
@joyously: What if the theme designer wants to have two different kinds of caption alignments, left and center? In that case [multiple block styles per block](https://github.com/WordPress/gutenberg/issues/14598) should be supported, so not only the general style of image (and its caption) but also its additional caption alignment context is selectable for the user.

**Comment by joyously** [comment]
> What if the theme designer wants to have two different kinds of caption alignments, left and center? In that case multiple block styles per block should be supported, 

I don't agree that the user should select it per block or even per page. If the theme designer wants to provide choices, the theme designer should put a theme option in the Customizer. Giving the user too many options as a low level means that they can so easily make their site an inconsistent mess, and changing their mind involves re-editing each page. I say let the theme do its job of keeping things consistent and don't put too many options into the editor. (and that goes for Full Site Editing times 2)

**Comment by strarsis** [comment]
@joyously: I agree. I always cringe when I see a page builder with padding/margin options for each individual element. Not only does this mean that the user has to set it up, it also means that these settings have to be consistently applied among all the pages.

**Comment by sarahannnicholson** [comment]
+1 for adding alignment to the caption toolbar

**Comment by nonobio** [comment]
Hello,

I added this css code found somewhere else, it works for me:

`.wp-block-image .aligncenter>figcaption, .wp-block-image .alignleft>figcaption, .wp-block-image .alignright>figcaption, .wp-block-image.is-resized>figcaption {
    text-align: center;
}`

I miss also style since gutenberg.
With classic editor, caption text was centered, with a smaller font and a grey background, so it was well differentiable from the normal text.

Now, with the new gutenberg wordpress editor, caption text is the same font than normal text, not aligned and with no background (i only succeeded to align the text with the CSS custom above).

Just to be sure: is it the same for everybody (miss caption alignement and style) or does it depend of the theme used ?

**Comment by paaljoachim** [comment]
Hi @nonobio 

I just tested with the Theme Twenty Twenty One. Which can be downloaded from here. https://github.com/WordPress/twentytwentyone

I added an image with a caption to see where the caption is located by default.
On Large or Full Size it aligned to the bottom center.
![Screen Shot 2020-11-25 at 16 43 48](https://user-images.githubusercontent.com/5323259/100250031-807b3200-2f3d-11eb-9cc9-eb35144f2941.png)

Upon Medium or Thumbnail it did not, and I will add an issue for it on the Github repo for the theme.
https://github.com/WordPress/twentytwentyone/issues/890

**Comment by nonobio** [comment]
Hi :)

Thanks for the test, i decovered that i could also test myself with "preview" theme ! :)

So i did : you're right : caption is aligned to the center without custom CSS with Twenty Twenty theme.

And the font is also a little different compared to the body text.

But no grey background but it is'nt a big deal.

I will contact my theme support or try to find css custom to apply a different font to caption text.

If you have css code or plugin about customizing captions text it will help.

Thanks a lot :)

**Comment by nonobio** [comment]
Hi :)

I found a solution to align and styling my captions with gutenberg editor like it was on classic editor (on my theme).
I analyzed font and color of my old captions (with chrome add-on Font Finder) then i applied this CSS code:

`.wp-block-image figcaption {
color: #888888;
background-color: #f4f4f4;
font-size: 13px;
text-align: start;
font-weight: 600;
margin: 0px;
padding: 10px
}`

I don't know anything in coding, i just tried and checked with Customizer\CSS to find the good solution.

All seems to work but tell me if you see something wrong with my code.

Thanks :)

**Comment by ellatrix** [comment]
IMO there should not be an option for this for individual captions. Maybe a global style option?

**Comment by paaljoachim** [comment]
I do think some images one might want the Image caption on the right, most in the center and others to the left. To create some design. But one could perhaps have a default global style option.

**Comment by paaljoachim** [comment]
**Image caption styles global styles support**
https://github.com/WordPress/gutenberg/issues/27480

**Comment by westonruter** [comment]
I'm closing this favor of the older issue:

* https://github.com/WordPress/gutenberg/issues/12997
